test/rules/param-property-in-constructor/test.ts.lint

```diff
@@ -0,0 +1,65 @@
+class FailingExample1 {
+    constructor(fruit: string, public num: number, private animal: string, protected mineral: string) {
+        //Both usage here are OK since `fruit` is a regular parameter and not a class property
+        fruit = "banana";
+        this.fruit = "persimmon";
+
+        this.num = 10;
+        num = 5;
+        ~~~ [The constructor parameter 'num' should only be accessed with the 'this' keyword: 'this.num']
+
+        animal = "aardvark";
+        ~~~~~~ [The constructor parameter 'animal' should only be accessed with the 'this' keyword: 'this.animal']
+        this.animal = "skunk";
+
+        mineral = "Gypsum";
+        ~~~~~~~ [The constructor parameter 'mineral' should only be accessed with the 'this' keyword: 'this.mineral']
+        this.mineral = "Fluorite";
+    }
+}
+
+class FailingExample2 {
+    constructor(public thing: Object) {
+        const wow=thing.doTheThing()
```

Comment:
yeah that was my thinking with the way it was written initially.  I was thinking that if we just require `this.` on everything, then there is consistency for how all constructor params are used, regardless if they are used as properties or not.